### DISCUSSION TITLE
ST_Transform on parameter geometry

### DISCUSSION THREAD

**Discussion by Simonkampendahl** [NetTopologySuite/NetTopologySuite#734] [Q&A]
i have a database where the geometry is in SRID 3006.
And i want to support query with different SRID.

Before using entity framework i could simply use ST_Transform(geom, srid) with ST_Intersects to search my database geometries with any srid.

using `.Where(f => f.Geometry.Intersects(geom)),` works great if its the same SRID.

i tried doing:
`.Where(f => f.Geometry.Intersects(EF.Functions.Transform(geom, 3006)))`
but this crashes.

i tried doing:
`.Where(f => EF.Functions.Transform(f.Geometry, geom.SRID).Intersects(geom))`
Now this does seem to work (atleast the debugged sql query string looks right). 
`-- @__geom_SRID_1='4326'
-- @__geom_2='POINT (15.51 61.59)' (DbType = Object)
SELECT *
FROM myTable AS r
WHERE ST_Intersects(ST_Transform(r.geom, @__geom_SRID_1), @__geom_2)`
but because it needs to convert the SRID for every row in the database it's extremly slow and times out.

Is there no way to add ST_Transform on my input geometry?

**Comment by FObermaier** [comment]
You need to add some reprojection library to your project and reproject your input point(s) before passing it to EF.
You could use e.g.
* [ProjNet](https://www.nuget.org/packages/ProjNet)
* [SharpProj](https://www.nuget.org/packages/SharpProj.NetTopologySuite)
* [DotSpatial.Projections](https://www.nuget.org/packages/DotSpatial.Projections)
* [GDAL](https://www.nuget.org/packages/GDAL)'s OSR package

**Reply by Simonkampendahl** [reply]
Is there a reason it cant be done using for example EF.Functions.Transform?
It seems necessary to need the overhead of those projects when the database already have the functionality built in?

I was able to solve my without any extra library by doing:
```
_dbContext
  .MyTable
  .FromSqlRaw("select * from MyTable where ST_Intersects(geom, ST_Transform(ST_SetSRID(@p0, @p1), 3006))", geom, geom.SRID)
  .AsNoTracking()
  .ToListAsync()
```

This sets the geom in query to the geom.SRID, then it transforms it to the same SRID as the database table geometry is using.
But obviously i'd prefer to not need raw sql command.
